### PR TITLE
Build OnnxRT with Openvino EP on Windows

### PR DESCRIPTION
**Pull request by avidiyal**
OnnxRT with Openvino EP on Windows

Cmake and code changes to support both Linux and Windows OS
Build instructions updated to support windows OS

### CODE REVIEW

File: cmake/onnxruntime_providers.cmake [hunk 1]
```diff
@@ -112,88 +112,6 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/providers/cp
 set_target_properties(onnxruntime_providers PROPERTIES LINKER_LANGUAGE CXX)
 set_target_properties(onnxruntime_providers PROPERTIES FOLDER "ONNXRuntime")
 
-
-if (onnxruntime_USE_MIMALLOC)
```
Comment (jywu-msft) [inline cmake/onnxruntime_providers.cmake]: This was removed by accident?

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -518,9 +518,9 @@ endif()
 
 if(onnxruntime_USE_OPENVINO)
 
-    if(WIN32)
-        message(FATAL_ERROR "Building on Windows using OpenVINO execution provider is not supported")
-    endif()
+  #  if(WIN32)
```
Comment (jywu-msft) [inline cmake/CMakeLists.txt]: remove these lines instead of commenting out.

File: BUILD.md [hunk 3]
```diff
@@ -219,26 +219,44 @@ ONNX Runtime supports OpenVINO Execution Provider to enable deep learning infere
...
 
 - Build ONNX Runtime using the below command.
 
+  For Linux:
+  
    <code>./build.sh --config RelWithDebInfo --use_openvino <hardware_option>  </code>
 
+  For Windows:
+  
+  <code> build.bat --config Release --cmake_generator "Visual Studio 16 2019" --use_openvino <hardware_option> --build_shared_lib --parallel</code>
```
Comment (jywu-msft) [inline BUILD.md]: Why do the instructions have --cmake_generator "Visual Studio 16 2019"  , isn't the default generator (VS 2017) sufficient?
also, neither --build_shared_lib nor --parallel are required right?  ideally, we should have the same build options as for Linux above.

File: onnxruntime/core/providers/openvino/openvino_graph.cc [hunk 4]
```diff
@@ -11,6 +11,11 @@
 #include <Python.h>
 
 #include <inference_engine.hpp>
+#ifdef OPTIONAL
```
Comment (jywu-msft) [inline onnxruntime/core/providers/openvino/openvino_graph.cc]: add a comment as to why OPTIONAL has to be undef'd is being done.

File: onnxruntime/core/providers/openvino/openvino_graph.cc [hunk 5]
```diff
@@ -326,7 +332,7 @@ size_t OpenVINOGraph::DeduceBatchSize(Ort::CustomOpApi ort, const OrtValue* inpu
 
 // Starts an asynchronous inference request for data in slice indexed by batch_slice_idx on
 // an Infer Request indexed by infer_req_idx
-void OpenVINOGraph::StartAsyncInference(Ort::CustomOpApi ort, const OrtValue* input_tensors[],
+void OpenVINOGraph::StartAsyncInference(Ort::CustomOpApi ort, std::vector<const OrtValue*> input_tensors,
```
Comment (jywu-msft) [inline onnxruntime/core/providers/openvino/openvino_graph.cc]: can you give some background on this and other api signature changes?
now we want to use a vector for inputs and outputs and pass the vector by value? 
why is this needed for enabling windows build?

File: BUILD.md (hunk 3 above)
Comment (avidiyal) [reply BUILD.md]: VSC++ does not support variable length array and was throwing build errors, so we had to use vector which passed windows build

File: onnxruntime/core/providers/openvino/openvino_graph.cc (hunk 5 above)
Comment (avidiyal) [reply onnxruntime/core/providers/openvino/openvino_graph.cc]: MSVC++ does not support variable length array and was throwing build errors, so we had to use vector which passed windows build

File: BUILD.md (hunk 3 above)
Comment (avidiyal) [reply BUILD.md]: Currently we tested it on 2019. We could not download older version of Visual studio 2017

File: BUILD.md [hunk 6]
```diff
@@ -219,26 +219,42 @@ ONNX Runtime supports OpenVINO Execution Provider to enable deep learning infere
...
+(https://docs.openvinotoolkit.org/2019_R1.1/_docs_install_guides_installing_openvino_windows.html#usb-myriad (Windows))
+
+- To configure Intel<sup>®</sup> Vision Accelerator Design based on 8 Movidius<sup>TM</sup> MyriadX VPUs, please follow the configuration guide from (https://docs.openvinotoolkit.org/2019_R1.1/_docs_install_guides_installing_openvino_linux.html#install-VPU (Linux))
+(https://docs.openvinotoolkit.org/2019_R1.1/_docs_install_guides_installing_openvino_windows.html#hddl-myriad (Windows))
 
 
 - Build ONNX Runtime using the below command.
 
+  For Linux & Windows:
+
```
Comment (jywu-msft) [inline BUILD.md]: for windows it would be build.bat

Comment (avidiyal) [reply BUILD.md]: Thank you George. Updated it now
